### PR TITLE
Zwave db rework

### PR DESCRIPTION
**Pull request by Bounz**

### CODE REVIEW

File: BaseFiles/Common/html/ext/zwave/_nodesetup.js [hunk 1]
```diff
@@ -543,21 +581,21 @@ HG.WebApp.GroupModules.SensorMultiLevelParameterGet = function () {
 function zwave_AssociationGet(nodeid, groupid, callback) {
     $.get('/' + HG.WebApp.Data.ServiceKey + '/HomeAutomation.ZWave/' + nodeid + '/Association.Get/' + groupid + '/', function (data) {
         if (typeof callback != 'undefined' && callback != null) {
-            callback(data.ResponseValue);
```
Comment (genielabs) [inline BaseFiles/Common/html/ext/zwave/_nodesetup.js]: Please keep back the use of ResponseText object. See also my other note:
https://github.com/genielabs/mig-service-dotnet/pull/7#discussion_r56712929

**Comment by genielabs** [comment]
Hi BounZ, there's a javascript error when the device is not found in Pepper1 DB:

`
"{"ClassName":"System.IO.FileNotFoundException","Message":"Could not find file \"/usr/local/bin/homegenie/additionalZwaveDevices.xml\".","InnerException":null,"HelpURL":null,"StackTraceString":"  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean anonymous, FileOptions options) [0x00000] in <filename unknown>:0 \n  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share) [0x00000] in <filename unknown>:0 \n  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)\n  at System.IO.FileInfo.Open (FileMode mode, FileAccess access, FileShare share) [0x00000] in <filename unknown>:0 \n  at System.IO.FileInfo.Open (FileMode mode, FileAccess access) [0x00000] in <filename unknown>:0 \n  at System.IO.FileInfo.OpenText () [0x00000] in <filename unknown>:0 \n  at (wrapper remoting-invoke-with-check) System.IO.FileInfo:OpenText ()\n  at MIG.Interfaces.HomeAutomation.Pepper1Db.GetDeviceInfoInDb (System.String filename, System.String manufacturerId, System.String version) [0x00000] in <filename unknown>:0 \n  at MIG.Interfaces.HomeAutomation.Pepper1Db.GetDeviceInfo (System.String manufacturerId, System.String version) [0x00000] in <filename unknown>:0 \n  at MIG.Interfaces.HomeAutomation.ZWave.InterfaceControl (MIG.MigInterfaceCommand request) [0x00000] in <filename unknown>:0 \n  at MIG.MigService.Gateway_PreProcessRequest (System.Object sender, MIG.ProcessRequestEventArgs args) [0x00000] in <filename unknown>:0 ","RemoteStackTraceString":null,"RemoteStackIndex":0,"HResult":-2146232799,"Source":"mscorlib","ExceptionMethod":null,"Data":null,"FileNotFound_FileName":"/usr/local/bin/homegenie/additionalZwaveDevices.xml","FileNotFound_FusionLog":null}"
`

also if there's data files to store, please keep them in the same ZWaveLib.dll folder as it happens with *zwavenodex.xml* file.

**Comment by Bounz** [comment]
Hi, Gene.
The error is because I forgot to include that file (additionalZwaveDevices.xml) into csproj.
This is the database of ZWave devices which are not present at pepper1, but were in HG's json files. So, this database is distributed with HG and should not be backed up.
Also, I included this file in the HG project, not in Mig.HomeAutomation, because it simplifies distribution of Mig.HomeAutomation.dll library.

**Comment by genielabs** [comment]
it's ok to include the file in HG project, but read my other comment here:

https://github.com/genielabs/mig-service-dotnet/pull/9#issuecomment-202351219

**Pull request merged**